app/validators/special_character_validator.rb

```diff
@@ -1,17 +1,17 @@
 class SpecialCharacterValidator < ActiveModel::EachValidator
 
   def validate_each(record, attribute, value)
-    return if value.nil?
+    return if value.nil? || value.blank?
 
-    if allow_comma == true && allow_numbers == true
-      regex = /^[a-zA-Z0-9 ,'-]+$/i
-    elsif allow_comma == true
-      regex = /^[a-zA-Z ,'-]+$/i
-    elsif allow_numbers == true
-      regex = /^[a-zA-Z0-9 '-]+$/i
-    else
-      regex = /^[a-zA-Z '-]+$/i
-    end
+    regex = if allow_comma == true && allow_numbers == true
```

Comment:
This is not very scalable - having an if condition for each combination of options.

You could start off with a regex and add to it based on the conditions ?